.github/workflows/tests-levels-matrix.php

```diff
@@ -0,0 +1,32 @@
+<?php declare(strict_types = 1);
+
+shell_exec('php vendor/bin/phpunit --list-tests-xml test-list.xml');
+
+$simpleXml = simplexml_load_file('test-list.xml');
+if ($simpleXml === false) {
+	throw new RuntimeException('Error loading test-list.xml');
+}
+
+$testFilters = [];
+foreach($simpleXml->testCaseClass as $testCaseClass) {
+	foreach($testCaseClass->testCaseMethod as $testCaseMethod) {
+		if ((string) $testCaseMethod['groups'] !== 'levels') {
+			continue;
+		}
+
+		$testCaseName = (string) $testCaseMethod['id'];
+
+		[$className, $testName] = explode('::', $testCaseName, 2);
+		$fileName = 'tests/'. str_replace('\\', DIRECTORY_SEPARATOR, $className) . '.php';
+
+		$filter = str_replace('\\', '\\\\', $testCaseName);
+
+		$testFilters[] = sprintf("%s --filter '%s'", $fileName, $filter);
```

Comment:
Why is doubling backslashes needed? I’d say that instead of that we need escapeshellarg() for both fileName and filter.

Reply:
Running the plain command on the CLI only works only with double slashes

Reply:
I guess we can add escape_shellarg, but we still need this manual escaping

Reply:
lets give it a try and we will see

Reply:
https://github.com/phpstan/phpstan-src/actions/runs/7944377066/job/21689826256 shows that it doesn't work when we only use `escapeshellarg`.

Reply:
Why is that? I noticed it doesn't work but I don't know what eats the characters, whether it's bash or PHP. Doubling the backslashes seems very peculiar to me, maybe there are other things we need to fix too but we're currently unaware of that.

Reply:
Phpunit examples in docs also mention double backslashes

https://docs.phpunit.de/en/9.6/textui.html#textui-examples-filter-patterns

Reply:
Its interpreted as regex. Let mit try preg_quote()

Reply:
I know. It's probably a bash thing.

Reply:
preg_quote would escape the following characters: `. \ + * ? [ ^ ] $ ( ) { } = ! < > | : - #`

the only one of these which can show up in a class/method name is `\`. I think we can use `preg_quote` if you prefer but it should be enough to escape the backslash IMO